### DISCUSSION TITLE
Any advice on builder a web api for Quartz.Net

### DISCUSSION THREAD

**Discussion by Sicos1977** [quartznet/quartznet#1602] [Q&A]
Hi @lahma  ,

Because I needed a replacement for .NET remoting (that forced me to use .NET 4.8) I started building a web api(controller) for Quartz.Net. You can find that over here --> https://github.com/Sicos1977/QuartzWebApi . I just started a few days ago so please don't look at classes and controllers I made here and there.

But still the question if you can have a look to see if I'm on the right way and if you have any tips or tricks then please let me know

Greetings,
@Sicos1977

**Comment by lahma** [comment]
I hope we don't get into overlapping paths here. I currently don't have anything in the work w.r.t. HTTP based API, but some others might have:

* https://github.com/quartznet/quartznet/issues/511#issuecomment-1120475228 @tonyqus is willing to help here (sorry for not pinging back sooner)
* @VilleHakli  had something in the works before, based on NET 6 minimal APIs, I believe this was early draft

Should we try to get something done as together or do we agree that @Sicos1977 will take the lead?

**Comment by Sicos1977** [comment]
At the moment I'm building the API in .NET 4.8 and after it is finished I'll switch the project to .NET Standard 2.0 to make it work in all versions of .NET. I also want to look if it is possible the make a version with websockets but that is something in the near feature.

One thing I still need to decide is how to host the API. I probably make it so that it can be hosted in IIS or self hosted.

**Comment by Sicos1977** [comment]
@lahma can you show me an example how this method would be used?

```c#
[HttpPost]
[Route("scheduler/pausejobs")]
public Task PauseJobs([FromBody] GroupMatcher<JobKey> matcher)
{
    //var match = GroupMatcher<JobKey>.GroupContains();
    return CreateScheduler.Scheduler.PauseJobs(matcher);
}
```

**Reply by lahma** [reply]
I'm not sure I understand the question. This would be called from scheduler proxy like in remoting case, using a HTTP client.

**Comment by Sicos1977** [comment]
I just mean how would you use this method from code?

How do you fill the GroupMatcher? I need to translate this to json but have no idea how the groupmatcher works

**Reply by lahma** [reply]
Maybe it would need separate DTO type to transfer the fields or a custom binder to get the actual matcher instance properly filled on web api logic side.

**Comment by tonyqus** [comment]
Rest Api Reference: https://github.com/nitinka/quartz-rest

**Comment by Sicos1977** [comment]
So far I added about 75% of all the needed methods. Last 2 weeks I had to do to many other things.

**Comment by Sicos1977** [comment]
Also want to add some kind of security layer so that a token can be used to access the webapi. But that is something for a next stage first need to get everything working.
